src/view/UserphraseView.cpp

```diff
@@ -75,6 +79,12 @@ void UserphraseView::addPhrase(int result)
     }
 
     emit model()->add(phrase, bopomofo);
+    
+    if (dialogType_ == DIALOG_MODIFY) {
+	if(model()->getAddResult() <= 0) {
```

Comment:
`model()->sourceModel()->getAddResult()` and revert all changes in `UserphraseSortFilterProxyModel.*`